streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1009,14 +1018,18 @@ private void addClientAssignments(final Map<String, Assignment> assignment,
                 AssignorError.NONE.code()
             );
 
-            if (encodeNextRebalanceTime) {
+            if (tasksRevoked) {
+                // TODO: once KAFKA-9821 is resolved we can leave it to the client to trigger this rebalance
+                log.debug("Requesting followup rebalance be scheduled immediately due to tasks changing ownership.");
+                info.setNextRebalanceTime(0L);
```

Comment:
This encodes an immediate followup rebalance for the consumer who would be receiving the revoked tasks. The consumer that is doing the revoking will also request a followup rebalance (in the client code) but due to KAFKA-9821 this cannot be relied upon when static membership is used.